api-put-object-copy.go

```diff
@@ -38,7 +41,7 @@ func (c Client) CopyObject(bucketName string, objectName string, objectSource st
 	}
 
 	// Set copy source.
-	customHeaders.Set("x-amz-copy-source", objectSource)
+	customHeaders.Set("x-amz-copy-source", url.QueryEscape(objectSource))
```

Comment:
You would have to use urlEncodeName() here rather than url.QueryEscape the problem is perhaps because if "+" is a character in the objectSource name. Then it would be escaped improperly.